docs/production/deployment.md

```diff
@@ -378,3 +378,182 @@ that can be quite difficult to debug.  Be sure to declare your
 unexpectedly (e.g. pointing to a DNS name that you haven't configured
 with multiple IPs for your Zulip machine; sometimes this happens with
 IPv6 configuration).
+
+
+## System and deployment configuration
+
+The file `/etc/zulip/zulip.conf` is used to configure properties of
+the system and deployment; `/etc/zulip/settings.py` is used to
+configure the application itself.  The `zulip.conf` sections and
+settings are described below.
+
+### `[machine]`
+
+#### `puppet_classes`
+
+A comma-separated list of the Puppet classes to install on the server.
+The most common is **`zulip::profile::standalone`**, used for a
+stand-alone single-host deployment.
+[Components](../overview/architecture-overview.html#components) of
+that include:
+ - **`zulip::profile::app_frontend`**
+ - **`zulip::profile::memcached`**
+ - **`zulip::profile::postgresql`**
+ - **`zulip::profile::redis`**
+ - **`zulip::profile::rabbitmq`**
+
+If you are using a [Apache as a single-sign-on
+authenticator](../production/authentication-methods.html#apache-based-sso-with-remote-user),
+you will need to add **`zulip::apache_sso`** to the list.
+
+#### `pgroonga`
+
+Set to the string `enabled` if enabling the [multi-language PGroonga
+search
+extension](../subsystems/full-text-search.html#an-optional-full-text-search-implementation).
+
+
+
+### `[deployment]`
+
+#### `deploy_options`
+
+Options passed by `upgrade-zulip` and `upgrade-zulip-from-git` into
+`upgrade-zulip-stage-2`.  These might be any of:
+
+ - **`--skip-puppet`** skips doing Puppet/apt upgrades.  The user will need
+   to run `zulip-puppet-apply` manually after the upgrade.
+ - **`--skip-migrations`** skips running database migrations.  The
+   user will need to run `./manage.py migrate` manually after the upgrade.
+ - **`--skip-purge-old-deployments`** skips purging old deployments;
+   without it, only deployments with the last two weeks are kept.
+
+Generally installations will not want to set any of these options; the
+`--skip-*` options are primarily useful for reducing upgrade downtime
+for servers that are upgraded frequently by core Zulip developers.
+
+#### `git_repo_url`
+
+Default repository URL used when [upgrading from a Git
+repository](../production/upgrade-or-modify.html#upgrading-from-a-git-repository).
+
+
+
+### `[application_server]`
+
+#### `http_only`
+
+If set to non-empty, [configures Zulip to allow HTTP
+access][using-http]; use if Zulip is deployed behind a reverse proxy
+that is handling SSL/TLS termination.
+
+#### `nginx_listen_port`
+
+Set to the port number if you [prefer to listen on a port other than
+443](#using-an-alternate-port).
+
+#### `no_serve_uploads`
+
+Set to true to configure `nginx` to enable [the S3 uploads
+backend](../production/upload-backends.html#s3-backend-configuration).
```

Comment:
The wording before was intentional; from my read, setting `no_serve_uploads` is required to do S3 uploads, but is not itself sufficient.  I don't want people thinking that setting it is all that is necessary.

I've also been say "non-empty" since if you set to the string "false" it's still actually enabled.

Reply:
Yeah, I guess I don't think we should tell people to specify any non-empty value.

I think we should the `false` behavior to be a bug and perhaps change that behavior?  Though I have this vague feeling that there's some backwards-compatible history here.

Reply:
I think that rabbit hole may not be worth going down, so I just added 3a4b38d7327e6be5687de1b5fc7197d84a219887 to make this a bit longer but at least clear.